hack/git/hooks/pre-commit

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+LC_ALL=C
+
+# e.g. "ARO-1234/short-description" or "hotfix/v20240321.00"
+valid_branch_regex="^(ARO-[0-9]{4}|hotfix)\/[a-z0-9._-]+$"
```

Comment:
The proposal looks good to me, but it seems like we now are requiring a JIRA for all PRs. This seems like a barrier for quick fix PRs, or contributors outside of Red Hat. Do you have thoughts on how we can do something like this without requiring a JIRA link?

Maybe we can add a regex for `nojira`: 
```suggestion
valid_branch_regex="^(nojira|ARO-[0-9]{4}|hotfix)\/[a-z0-9._-]+$"
```

Reply:
We should probably include an update here in this PR as well: https://github.com/Azure/ARO-RP/blob/master/README.md#contributing

Reply:
Along the same lines why not just:
```suggestion
valid_branch_regex="[a-z0-9._-]*\/[a-z0-9._-]+$"
```

As stated in the doc we just need difference and the names don't matter so why not just keep it open? I'm usually a fan of consistency but being simple seems even nicer here.

Reply:
I liked the idea of using github usernames as a prefix for this (pointed out in the doc), because the core thing we want to avoid is people squashing other IC's commits with a --force (unfortunately required in our repo today without a simple fix). I had to close this branch to match that new schema.

The point on open source contribs I was originally on-board with; that made more sense in a fork world, but now we're in a complex situation - do we plan to open up `create branch` permissions to the internet on this repo?

Reply:
I like this discussion let's move it to https://github.com/Azure/ARO-RP/pull/3479 (sorry; I would have preferred to keep this one open under a new branch but apparently that's not possible in GH or I messed up the flow)